src/api/EscargotPublic.cpp

```diff
@@ -102,24 +105,175 @@ inline AtomicString toImpl(AtomicStringRef* v)
     return AtomicString::fromPayload(reinterpret_cast<void*>(v));
 }
 
+class PlatformBridge : public Platform {
+public:
+    PlatformBridge(PlatformRef* p)
+        : m_platform(p)
+    {
+    }
+
+    virtual ~PlatformBridge()
+    {
+        // delete PlatformRef, so we don't care about PlatformRef's lifetime
+        delete m_platform;
+    }
+
+    virtual void* onMallocArrayBufferObjectDataBuffer(size_t sizeInByte) override
+    {
+        return m_platform->onMallocArrayBufferObjectDataBuffer(sizeInByte);
+    }
+
+    virtual void onFreeArrayBufferObjectDataBuffer(void* buffer, size_t sizeInByte) override
+    {
+        m_platform->onFreeArrayBufferObjectDataBuffer(buffer, sizeInByte);
+    }
+
+    virtual void* onReallocArrayBufferObjectDataBuffer(void* oldBuffer, size_t oldSizeInByte, size_t newSizeInByte) override
+    {
+        return m_platform->onReallocArrayBufferObjectDataBuffer(oldBuffer, oldSizeInByte, newSizeInByte);
+    }
+
+    virtual void markJSJobEnqueued(Context* relatedContext) override
+    {
+        // Job should be enqueued in the main thread
+        ASSERT(Global::inMainThread());
```

Comment:
IMHO we don't need to specify main-thread
Every thread can be act as main-thread.
And worker threads can use promise too

Reply:
Fixed it.
Regarding job queue, I'll separate it in later patch.